heron/healthmgr/src/java/org/apache/heron/healthmgr/policy/ToggleablePolicy.java

```diff
@@ -0,0 +1,136 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.heron.healthmgr.policy;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.logging.Logger;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import com.microsoft.dhalion.core.Action;
+import com.microsoft.dhalion.core.Diagnosis;
+import com.microsoft.dhalion.core.Measurement;
+import com.microsoft.dhalion.core.Symptom;
+import com.microsoft.dhalion.policy.HealthPolicyImpl;
+
+import org.apache.heron.api.generated.TopologyAPI;
+import org.apache.heron.healthmgr.HealthPolicyConfig;
+import org.apache.heron.healthmgr.common.PhysicalPlanProvider;
+
+import static org.apache.heron.healthmgr.HealthPolicyConfig.CONF_POLICY_ID;
+
+/**
+ * This Policy class
+ * 1. has a toggle switch
+ * 2. works with runtime config
+ */
+public class ToggleablePolicy extends HealthPolicyImpl {
+  private static final Logger LOG =
+      Logger.getLogger(ToggleablePolicy.class.getName());
+
+  protected PhysicalPlanProvider physicalPlanProvider;
+  protected String policyId;
+  private String policyIdRuntime;
+  protected HealthPolicyConfig policyConfig;
+
+  protected PolicyMode policyMode;
+
+  public enum PolicyMode {
+    activated,
+    deactivated
+  }
+
+  @Inject
+  public ToggleablePolicy(
+      @Named(CONF_POLICY_ID) String policyId,
+      HealthPolicyConfig policyConfig,
+      PhysicalPlanProvider physicalPlanProvider) {
+    this.physicalPlanProvider = physicalPlanProvider;
+    this.policyId = policyId;
+    this.policyConfig = policyConfig;
+
+    policyMode = policyConfig.getPolicyMode();
+    policyIdRuntime = policyId + ":runtime";
+  }
+
+  @Override
+  public Collection<Measurement> executeSensors() {
+    for (TopologyAPI.Config.KeyValue kv
+        : physicalPlanProvider.get().getTopology().getTopologyConfig().getKvsList()) {
+      LOG.fine("kv " + kv.getKey() + ":" + kv.getValue());
+      if (kv.getKey().equals(policyIdRuntime)) {
+        String val = kv.getValue();
+        if (PolicyMode.deactivated.name().equals(val)
```

Comment:
Can we first convert the `val` to `PolicyMode` type then compare it with the current `policyMode`? It's more succinct and easier to understand.

Reply:
updated